src/database/test/DatabaseTests.cpp

```diff
@@ -353,14 +353,25 @@ TEST_CASE("schema test", "[db]")
 class SchemaUpgradeTestApplication : public TestApplication
 {
   public:
-    SchemaUpgradeTestApplication(VirtualClock& clock, Config const& cfg)
+    using PreUpgradeFunc = std::function<void(SchemaUpgradeTestApplication&)>;
+
+  private:
+    PreUpgradeFunc const preUpgradeFunc;
+
+  public:
+    SchemaUpgradeTestApplication(VirtualClock& clock, Config const& cfg,
+                                 PreUpgradeFunc _preUpgradeFunc)
```

Comment:
Nitpick: this isn't mentioned in the code-style conventions, but we don't typical use the "prefix-underscore" because it isn't necessary with the "prefix-m" member variable convention

Reply:
Fixed.